src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,178 @@
+package core.basesyntax;
+
+import core.basesyntax.exception.InvalidUserCredentialsException;
+import core.basesyntax.exception.InvalidUserObjectException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    public void register_nullUser_should_throw_invalidUserException_negative() {
+        User user = null;
+        Assertions.assertThrows(InvalidUserObjectException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_userAlreadyExists_should_throw_invalidUserException_negative() {
+        User user = new User();
+        user.setLogin("test-mail1@gmail.com");
+        user.setPassword("test-password1");
+        user.setAge(22);
+        registrationService.register(user);
+        Assertions.assertThrows(InvalidUserObjectException.class,
+                () -> registrationService.register(user));
```

Comment:
The test `register_userAlreadyExists_should_throw_invalidUserException_negative` expects an `InvalidUserObjectException`, but the logic in `RegistrationServiceImpl` should throw an `InvalidUserCredentialsException` for an existing user. Update the test to expect the correct exception type.